jcapi.go

```diff
@@ -240,26 +278,31 @@ func (jc JCAPI) extractStringArray(input []interface{}) []string {
 	return returnVal
 }
 
-func getStringOrNil(input interface{}) string {
-	returnVal := ""
-
+func getStringOrNil(input interface{}) (s string) {
 	switch input.(type) {
 	case string:
-		returnVal = input.(string)
+		s = input.(string)
 	}
 
-	return returnVal
+	return
 }
 
-func getUint16OrNil(input interface{}) uint16 {
-	var returnVal uint16
-
+func getUint16OrNil(input interface{}) (i uint16) {
 	switch input.(type) {
 	case uint16:
-		returnVal = input.(uint16)
+		i = input.(uint16)
 	}
 
-	return returnVal
+	return
+}
+
+func getIntOrNil(input interface{}) (i int) {
+	switch input.(type) {
+	case int:
+		i = input.(int)
+	}
+
+	return
 }
```

Comment:
There's lots of changes already so I wouldn't bother with this now, but it just occurred to me this can be written much more terse using the non-panic version of type conversion if we want:
```
func getIntOrNil(input interface{}) (i int) {
    i, _ = input.(int)
    return
}
```
`input` will either type convert into `i`, or it will be left at the zero value as it is now.